docs/interpreter.md

```diff
@@ -6,10 +6,11 @@
 [StableHLO programs](spec_draft.md#programs) are computations over tensors
 (n-dimensional arrays), which, in the current model, are implemented using class
 `Tensor`. The underlying storage class for a `Tensor` object, `detail::Buffer`,
-stores the `mlir::ShapedType` of the tensor along with a contiguous byte array
-representing its data laid out in
-[major-to-minor order](https://www.tensorflow.org/xla/shapes). `detail::Buffer`
-objects are reference-counted to simplify memory management.
+stores the `mlir::ShapedType` of the tensor along with a
+`mlir::HeapAsmResourceBlob` object representing a mutable blob of tensor
+data laid out as contiguous byte array in
+[major-to-minor order](https://www.tensorflow.org/xla/shapes).
+`detail::Buffer` objects are reference-counted to simplify memory management.
```

Comment:
Some ownership / lifetime aspect should be documented: a Tensor is an immutable object and copying a tensor doesn't copy the underlying buffer?
(also the Tensor class could be documented, I don't see the header being modified here)